apps/aecore/src/aec_peer_connection.erl

```diff
@@ -1234,3 +1245,77 @@ get_micro_block_txs([TxHash | TxHashes], State, Acc) ->
         none         -> get_micro_block_txs(TxHashes, some, [TxHash | Acc]);
         {value, STx} -> get_micro_block_txs(TxHashes, State, [STx | Acc])
     end.
+
+pre_assembly_check(MicroHeader) ->
+    {ok, Hash} = aec_headers:hash_header(MicroHeader),
+    case aec_db:has_block(Hash) of
+        true -> known;
+        false ->
+            Validators = [fun validate_micro/1,
```

Comment:
I see this function runs check `aec_headers:validate_micro_block_cycle_time` that skips (!) check if prev hash not in DB. I assume following check `validate_connected_to_chain` does check that prev hash is in DB (it runs `aec_chain_state:hash_is_connected_to_genesis(aec_headers:prev_hash(...))` - I did not check that this actually checks that prev hash is in DB). So it looks fine but, as DB updates may be performed while this function runs, there is slight possibility that `validate_micro` check does not find prev hash in DB (hence passes without checking time) then concurrently prev hash is inserted in DB so when check `validate_connected_to_chain` runs it finds the prev hash. This leaves the timestamp not fully checked.

I consider this a minor issue though it hinders a bit readability. Any strong opinions on swapping `validate_micro` and `validate_connected_to_chain`?

Reply:
As it is now the timestamp will be checked also in `aec_chain_state:insert_block` (`assert_...`) - so no risk... But I have discussed with Tobias to later remove some redundant checks and then we should be careful not missing this check!